fedimint-api/src/db/mem_impl.rs

```diff
@@ -170,6 +176,21 @@ impl<'a> IDatabaseTransaction<'a> for MemTransaction<'a> {
 
         Ok(())
     }
+
+    fn rollback_tx_to_savepoint(&mut self) {
+        self.tx_data = self.savepoint.clone();
+
+        // Remove any pending operations beyond the savepoint
+        let removed_ops = self.num_pending_operations - self.num_savepoint_operations;
+        for _i in 0..removed_ops {
+            self.operations.pop();
+        }
+    }
+
+    fn set_tx_savepoint(&mut self) {
```

Comment:
OK, so after some discussion we decided that it's best to move forward with this. Backend will just use rocksdb all the time anyway.

Maybe just add some comments to document that we should avoid using these, especially on the client side, so if we ever decide so, we can easily stop using these.

Reply:
I added some comments in mem_impl.rs and the sled implementations that these shouldn't be used for production code as they don't implement MVCC correctly. Also clarified in db.rs that set_tx_savepoint ideally shouldn't be used on the client side.